crates/bevy_asset/src/loader.rs

```diff
@@ -307,6 +308,271 @@ pub enum DeserializeMetaError {
     DeserializeMinimal(SpannedError),
 }
 
+/// A builder for loading nested assets inside a `LoadContext`.
+pub struct NestedLoader<'a, 'b> {
```

Comment:
So I've played around with a type-state variant of this API. You can have a look at the Gist [here](https://gist.github.com/bushrat011899/6bd1b1173753475802cde3050249f6a3). Having experimented, I think the API you've gone for is better, since we don't really have the conflicting configuration options that typestate would resolve.

I would suggest pushing these builder types into their own file, since it's ~200 LOC with a very specific purpose that can be easily isolated from the rest of the `LoadContext`.